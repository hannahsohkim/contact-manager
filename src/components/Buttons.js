import React from 'react';
import EditForm from './EditForm'
import DeleteForm from './DeleteForm'
import { Button, Card } from 'semantic-ui-react'

class Buttons extends React.Component {
  constructor(props){
    super(props);
    this.toggleEdit = this.toggleEdit.bind(this);
    this.toggleDelete = this.toggleDelete.bind(this)
  }

  state = {
    editToggle: true,
    deleteToggle: true
  }

  toggleEdit() {
    this.setState(prevState => ({
      editToggle: !prevState.editToggle
    }))
  }

  toggleDelete() {
    this.setState(prevState => ({
      deleteToggle: !prevState.deleteToggle
    }))
  }

render() {
  const { toggleDelete, toggleEdit } = this;
  const { editToggle, deleteToggle } = this.state;

      if (editToggle && deleteToggle) {
         return (
          <Card.Content extra>
            <div className="ui two buttons">
              <Button
                className="test"
                onClick={toggleEdit}
                basic
                color='blue'>
              Edit
            </Button>
            <Button
              className="test2"
              onClick={toggleDelete}
              basic
              color='red'>
              Delete
            </Button>
            </div>
          </Card.Content>)
      } else {
        if (!editToggle) {
          return <EditForm id={this.props.id} toggle={toggleEdit} />
        }
        if (!deleteToggle) {
          return <DeleteForm id={this.props.id} toggle={toggleDelete} />
      }
    }
  }
}

export default Buttons;
