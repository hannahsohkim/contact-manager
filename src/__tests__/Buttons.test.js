import React from 'react';
import Enzyme, { shallow} from 'enzyme';
import Buttons from '../components/Buttons';
import renderer from 'react-test-renderer';

describe('Button Component', () => {
  test('renders', () => {
    const output = shallow(<Buttons />);
    expect(output.exists()).toBe(true);

    const tree = renderer
    .create(<Buttons id={'1'} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })

  // test('should handle the click event', () => {
  //   window.alert = jest.fn();
  //   const output = shallow(<Buttons />);
  //   output.find('.test').simulate('click');
  //   expect(window.alert).toHaveBeenCalledWith('clicked')
  // })

  test("editToggle should handle state changes", () => {
    const output = shallow(<Buttons />);

    expect(output.state().editToggle).toEqual(true);
    output.find('.test').simulate('click');
    expect(output.state().editToggle).toEqual(false);
  })

  test("deleteToggle should handle state changes", () => {
    const output = shallow(<Buttons />);

    expect(output.state().deleteToggle).toEqual(true);
    output.find('.test2').simulate('click');
    expect(output.state().deleteToggle).toEqual(false);
  })
})

