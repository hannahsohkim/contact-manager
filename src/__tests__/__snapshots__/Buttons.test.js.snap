// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders 1`] = `
<div
  className="extra content"
>
  <div
    className="ui two buttons"
  >
    <button
      className="ui blue basic button test"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="ui red basic button test2"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
